frontend/app/actions/ActionRefiners.scala

```diff
@@ -33,20 +42,73 @@ object Functions extends LazyLogging {
   def authenticated(onUnauthenticated: RequestHeader => Result = chooseRegister(_)): ActionBuilder[AuthRequest] =
     new AuthenticatedBuilder(AuthenticationService.authenticatedUserFor, onUnauthenticated)
 
-  def memberRefiner(onNonMember: RequestHeader => Result = notYetAMemberOn(_)) =
-    new ActionRefiner[AuthRequest, AnyMemberTierRequest] {
-      override def refine[A](request: AuthRequest[A]) = request.forMemberOpt {
-        _.map(member => MemberRequest(member, request)).toRight(onNonMember(request))
+  type SubRequestOrResult[A] = Future[Either[Result, SubReqWithSub[A]]]
+  type PaidSubRequestOrResult[A] = Future[Either[Result, SubscriptionRequest[A] with PaidSubscriber]]
+  type FreeSubRequestOrResult[A] = Future[Either[Result, SubscriptionRequest[A] with FreeSubscriber]]
+
+  type SubReqWithPaid[A] = SubscriptionRequest[A] with PaidSubscriber
```

Comment:
We could enrich those types (or create classes) so they provide accessor for a subscriber plan and tier, as they are used all over the place